src/overlays/actors/ovl_Obj_Shutter/z_obj_shutter.c

```diff
@@ -28,14 +28,65 @@ const ActorInit Obj_Shutter_InitVars = {
     (ActorFunc)ObjShutter_Draw,
 };
 
-#endif
+void ObjShutter_Init(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-extern UNK_TYPE D_060011E0;
+void ObjShutter_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Shutter/ObjShutter_Init.s")
+static u8 sScheduleScript[] = {
+    /* 0x0 */ SCHEDULE_CMD_CHECK_TIME_RANGE_S(10, 0, 20, 0, 0x9 - 0x6),
+    /* 0x6 */ SCHEDULE_CMD_RET_VAL_L(2),
+    /* 0x9 */ SCHEDULE_CMD_RET_VAL_L(1),
+};
+
+void ObjShutter_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    ObjShutter* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    ScheduleResult schedule;
+
+    Schedule_RunScript(globalCtx, sScheduleScript, &schedule);
+    if (schedule.result == 1) {
+        if (this->schedule != schedule.result) {
+            this->actor.velocity.y = 0.0f;
+        }
+        if ((this->velocity >= 80.0f) || (this->schedule == 0)) {
+            this->actor.velocity.y = 0.0f;
+            this->velocity = 80.0f;
+        } else {
+            this->velocity += 10.0f;
+        }
+    } else {
+        if (this->schedule != schedule.result) {
+            this->actor.velocity.y = 0.0f;
+        }
+        if (this->velocity != 0.0f) {
+            this->actor.velocity.y -= 3.0f;
+            this->velocity += this->actor.velocity.y;
+            if (this->velocity <= 0.0f) {
+                this->actor.velocity.y = fabsf(this->actor.velocity.y) * 0.8f;
+                if ((s32)this->actor.velocity.y == 0) {
+                    this->actor.velocity.y = 0.0f;
+                    this->velocity = 0.0f;
+                }
+            }
+        }
+    }
+    this->schedule = schedule.result;
+}
+
+void ObjShutter_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    ObjShutter* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Shutter/ObjShutter_Destroy.s")
+    Matrix_Translate(this->actor.world.pos.x, this->actor.world.pos.y + this->velocity, this->actor.world.pos.z,
+                     MTXMODE_NEW);
+    Matrix_Scale(0.1f, 0.1f, 0.1f, MTXMODE_APPLY);
+    Matrix_RotateYS(this->actor.world.rot.y, MTXMODE_APPLY);
+    OPEN_DISPS(globalCtx->state.gfxCtx);
```

Comment:
```suggestion
    Matrix_RotateYS(this->actor.world.rot.y, MTXMODE_APPLY);

    OPEN_DISPS(globalCtx->state.gfxCtx);
```
since it's a new block separate from the matrix manipulation.